scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -523,59 +525,88 @@ private void satisfyGoalGeneral(Goal goal) {
     //setting the number of conflicts detected at first evaluation, will be used at backtracking
     evaluation.forGoal(goal).setNbConflictsDetected(missingConflicts.size());
     assert missingConflicts != null;
-    boolean madeProgress = true;
 
+    final var itConflicts = missingConflicts.iterator();
+
+    //create new activity instances for each missing conflict
+    while (itConflicts.hasNext()) {
+      final var missing = itConflicts.next();
+      assert missing != null;
 
-    while (!missingConflicts.isEmpty() && madeProgress) {
-      madeProgress = false;
+      //determine the best activities to satisfy the conflict
+      if (!analysisOnly && (missing instanceof MissingActivityInstanceConflict missingActivityInstanceConflict)) {
+        final var acts = getBestNewActivities(missingActivityInstanceConflict);
+        assert acts != null;
```

Comment:
I do not know... to be sure we enforce the contract?

Reply:
I'm not sure I understand. Could you clarify what you mean by "to be sure we enforce the contract"? If you look at `getBestNewActivities`, on the second line we declare a linked list, which is the object that is returned, potentially with some activities added to it. `acts` can be empty, but there is no way for it to be `null`.

Reply:
I guess it's defensive programming, you protect against potential changes in other ways than tests. We can remove it if you prefer.

Reply:
Yeah, let's cut it 👍

Reply:
Sounds good to me - we started out with a lot of null-related bugs in the scheduler prototype because very many things were very often null. While integrating it into Aerie, we still haven't quite finished adapting the scheduler to the Aerie style of avoiding null altogether